### PR TITLE
integration of update and tile expire procedure

### PR DESCRIPTION
**Pull request by stevo01**
The procedures is based on strategies described on following web page:
see https://ircama.github.io/osm-carto-tutorials/updating-data/

Implements https://github.com/Overv/openstreetmap-tile-server/issues/2.

### CODE REVIEW

**Comment by stevo01** [comment]
verification of update procedure is difficult at the moment. 
The logfile of the update script (/var/log/tiles/run.log) shows "Osmosis error" and Osmosis log (/var/log/tiles/osmosis.log) shows download errors. I assume there is an error on server site "https://planet.openstreetmap.org/replication/minute" at the moment.

**Comment by stevo01** [comment]
following log file extract shows the progress og update procedure ...
```
[2019-06-16 13:32:01] 11458 start import from seq-nr 3540212, replag is 1 day(s) and 3 hour(s)
[2019-06-16 13:32:01] 11458 downloading diff
[2019-06-16 13:41:58] 11458 filtering diff
[2019-06-16 13:42:01] 11724 start import from seq-nr 3540272, replag is 1 day(s) and 2 hour(s)
[2019-06-16 13:42:01] 11724 downloading diff
[2019-06-16 13:55:19] 11724 filtering diff
[2019-06-16 13:56:01] 12039 start import from seq-nr 3540332, replag is 1 day(s) and 1 hour(s)
[2019-06-16 13:56:01] 12039 downloading diff
[2019-06-16 14:11:27] 12039 filtering diff
[2019-06-16 14:12:02] 12376 start import from seq-nr 3540392, replag is 1 day(s) and 1 hour(s)
[2019-06-16 14:12:02] 12376 downloading diff
[2019-06-16 14:14:14] 12376 [error] Osmosis error
[2019-06-16 14:14:14] 12376 resetting state
```

**Comment by Overv** [comment]
So to make sure I understand correctly, with these changes the update procedure comes down to overwriting the mapped `data.osm.pbf` and `data.poly` files with new data and then it will automatically start importing and rerendering any tiles once they are requested?

File: openstreetmap-tiles-update-expire [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+fi
+
+    seq=`cat $WORKOSM_DIR/state.txt | grep sequenceNumber | cut -d= -f2`
+
+    m_ok "start import from seq-nr $seq, replag is `osmosis-db_replag -h`"
+
+    /bin/cp $WORKOSM_DIR/state.txt $WORKOSM_DIR/last.state.txt
+    m_ok "downloading diff"
+
+    if ! $OSMOSIS_BIN --read-replication-interval workingDirectory=$WORKOSM_DIR --simplify-change --write-xml-change $CHANGE_FILE 1>&2 2> "$OSMOSISLOG"; then
```
Comment (stevo01) [inline openstreetmap-tiles-update-expire]: import diff files from osm server

File: openstreetmap-tiles-update-expire [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  m_ok "filtering diff skipped"
+fi
+    m_ok "importing diff"
+#------------------------------------------------------------------------------
+# Previously openstreetmap-tiles-update-expire tried to dirty layer
+# "$EXPIRY_MAXZOOM - 3" (which was 15) only.  Instead we write all expired
+# tiles in range to the list (note the "-" rather than ":" in the "-e"
+# parameter).
+#------------------------------------------------------------------------------
+    if ! $OSM2PGSQL_BIN -a --slim -e$EXPIRY_MINZOOM-$EXPIRY_MAXZOOM $OSM2PGSQL_OPTIONS -o "$EXPIRY_FILE.$$" $CHANGE_FILE 1>&2 2> "$PGSQLLOG"; then
```
Comment (stevo01) [inline openstreetmap-tiles-update-expire]: import diff file

File: openstreetmap-tiles-update-expire [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+#------------------------------------------------------------------------------
+# When expiring tiles we need to define the style sheet if it's not "default".
+# In this case it's "ajt".
+# Previously all tiles on the "dirty" list between $EXPIRY_MINZOOM and
+# $EXPIRY_MAXZOOM were dirtied.  We currently re-render
+# tiles >= $EXPIRY_MINZOOM and < $EXPIRY_DELETEFROM, expiry from 14 and
+# delete >= $EXPIRY_DELETEFROM and <= $EXPIRY_MAXZOOM.
+# The default path to renderd.sock is fixed.
+#------------------------------------------------------------------------------
+    if ! render_expired --map=ajt --min-zoom=$EXPIRY_MINZOOM --touch-from=$EXPIRY_TOUCHFROM --delete-from=$EXPIRY_DELETEFROM --max-zoom=$EXPIRY_MAXZOOM -s /var/run/renderd/renderd.sock < "$EXPIRY_FILE.$$" 2>&1 | tail -8 >> "$EXPIRYLOG"; then
```
Comment (stevo01) [inline openstreetmap-tiles-update-expire]: (re)render expired tiles / (update cached tiles)

**Comment by stevo01** [comment]
> So to make sure I understand correctly, with these changes the update procedure comes down to overwriting the mapped `data.osm.pbf` and `data.poly` files with new data and then it will automatically start importing and rerendering any tiles once they are requested?

The main logic of update process is controlled by the script openstreetmap-tiles-update-expire. Following steps will be performed:
1) read diff files from osm server. ( openstreetmap-tiles-update-expire, line 147)
2) optional step: filter out relevant data for given area spezified by *.poly file (file openstreetmap-tiles-update-expire, line 154)
3) import the diff files to psql database and produce file with list of expired tiles (file openstreetmap-tiles-update-expire, line 167)
4) (re)render expired tiles / update tile cache (file openstreetmap-tiles-update-expire, 190)

**Comment by Overv** [comment]
I've clarified the README based on your comments and merged it :) Thanks for the awesome contribution. :+1:

**Pull request merged**